File: bilfind_app/lib/models/comment_model.dart
class CommentModel {
  CommentModel({
    required this.id,
    required this.postId,
    required this.userId,
    this.parentId,
    required this.content,
    required this.createdAt,
    required this.ownerName,
    required this.ownerEmail,
    required this.ownerDepartment,
  });
  late final String id;
  late final String postId;
  late final String userId;
  late final String? parentId;
  late final String? ownerPhoto;
  late final String content;
  late final DateTime createdAt;
  late final String ownerName;
  late final String ownerEmail;
  late final String ownerDepartment;

  CommentModel.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    postId = json['postId'];
    userId = json['userId'];
    parentId = json['parentId'];
    ownerPhoto = json['ownerPhoto'];
    content = json['content'];
    createdAt = DateTime.parse(json['createdAt']).toLocal();
    ownerName = json['ownerName'];
    ownerEmail = json['ownerEmail'];
    ownerDepartment = json['ownerDepartment'];
  }
}

File: backend/src/models/comment-model.ts
import { Expose, Transform } from "class-transformer";
import { ObjectId } from "mongodb";
import { User } from "./user-model";
import { Mapper } from "../utils/mapper";
import { Departments } from "../utils/enums";

export class CommentModel {
  @Transform((value) => value.obj._id.toString())
  @Expose()
  _id?: ObjectId;

  @Expose()
  postId: string;

  @Expose()
  userId: string;

  @Expose()
  parentId?: string;

  @Expose()
  content: string;

  @Expose()
  createdAt: Date;

  @Expose()
  isDeleted: boolean;
}

export class CommentResponseDTO {
  @Expose()
  id: String;

  @Expose()
  postId: string;

  @Expose()
  userId: string;

  @Expose()
  parentId?: string;

  @Expose()
  content: string;

  @Expose()
  createdAt: Date;

  @Expose()
  ownerPhoto?: string;

  @Expose()
  ownerName: string;

  @Expose()
  ownerEmail: string;

  @Expose()
  ownerId: string;

  @Expose()
  ownerDepartment: Departments;
}

export const mapToCommentResponseDTO = (commentModel: CommentModel, user: User): CommentResponseDTO => {
  return Mapper.map(CommentResponseDTO, {
    ...commentModel,
    id: commentModel._id!.toString(),
    ownerPhoto: user.profilePhoto,
    ownerName: user.name + " " + user.familyName,
    ownerDepartment: user.departmant,
    ownerEmail: user.email,
    ownerId: user._id!.toString(),
  });
};
